mettle/src/network_server.c

```diff
@@ -83,12 +83,36 @@ int network_server_listen_tcp(struct network_server *ns,
 	ns->sin.sin6_port = htons((uint16_t)port);
 	ns->port = port;
 
-	if (host == NULL) {
+	if (host == NULL || strlen(host) == 0) {
```

Comment:
Down on line 136 there's a block like this:
```
#ifdef IPV6_V6ONLY
	int no = 0;
	setsockopt(ns->listener, IPPROTO_IPV6, IPV6_V6ONLY, (void *)&no, sizeof(no));
#endif
```

Which is always setting the `IPV6_V6ONLY` option to false / disabling it. It should be disabled (`no = 0`) when the host is not specified `(host == NULL || strlen(host) == 0)` or the host is an IPv4 address and you map it yourself to an IPv6 address.

If `host` is a true IPv6 address, then `IPV6_V6ONLY` should be enabled (`no = 1`) to ensure that you're only binding on IPv6 addresses. `no = 1` is silly to read, so I'd suggest renaming it something that makes sense like `v6_only` and setting it to 1/0 as necessary.